### DISCUSSION TITLE
Drop of Java 11 compatibility?

### DISCUSSION THREAD

**Discussion by alexandrebouchard** [eclipse-jdt/eclipse.jdt.core#2253] [General]
Our gradle CI started failing recently due to `org/eclipse/core/runtime/OperationCanceledException` in a gradle build which used to be compatible with Java 11 but is not longer (despite not changing the build script). Unfortunately, we have a system pinned at Java 11 for the foreseeable future. 

The error message is:

```
java.lang.UnsupportedClassVersionError: org/eclipse/core/runtime/OperationCanceledException 
has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```

We have not changed our build script dependencies, so I suspect a jar file on maven central might have been re-generated recently using a newer version of Java, losing compatibility with Java 11 as a side effect. The new one uses class version 61.0, making it incompatible with Java 11. If that's the case, would it be possible to regenerate to maintain support for Java 11? Thank you in advance for your help!

To reproduce:

```
git clone https://github.com/UBC-Stat-ML/blangSDK.git 
cd blangSDK
./gradlew build
```

**Comment by iloveeclipse** [comment]
> would it be possible to regenerate to maintain support for Java 11? Thank you in advance for your help!

No. JDT and rest of SDK require Java 17. If you need support of older Java runtimes, don't update your build tooling and pin all SDK dependencies to Java 11 compatible versions.

**Comment by alexandrebouchard** [comment]
Thanks for the quick response @iloveeclipse  The dependencies in my build file are all pinned, but I suppose something in the transitive closure might not. I will attempt to fix that dependency to a specified version manually, but it is not clear in what artifact "org/eclipse/core/runtime/OperationCanceledException" sits in. Is there a way to find this out (and the last version of that artifact that supports Java 11). Thanks again.
